BlueprintUI/Sources/Internal/LayoutModeKey.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+enum LayoutModeKey: EnvironmentKey {
+    static let defaultValue: LayoutMode = .default
+}
+
+extension Environment {
+    /// This mode will be inherited by descendant BlueprintViews that do not have an explicit
+    /// mode set.
+    var layoutMode: LayoutMode {
```

Comment:
I put this here for now to make the inheritance convenient, may wind up removing it from Environment so that we're not managing two sources of truth (this and RenderContext).